components/script/dom/htmlcollection.rs

```diff
@@ -232,7 +232,7 @@ impl HTMLCollection {
 
 // TODO: Make this generic, and avoid code duplication
```

Comment:
The eventual is actually to solve this TODO comment here. Using dynamic dispatch is a first step towards that direction.